docs/Using-SRIOV-and-Clear-Containers.md

```diff
@@ -0,0 +1,265 @@
+
+1. Add `vfio-pci` device driver
+
+   `vfio-pci` is a driver which is used to reserve a VF PCI device.  Add it:
+   ```
+   $ sudo modprobe vfio-pci
+   ```
+
+2. Find our NICs of interest
+
+   Find PCI details for the NICs in question:
+   ```
+   $ lspci | grep Ethernet
+   00:19.0 Ethernet controller: Intel Corporation Ethernet Connection I217-LM (rev 04)
+   01:00.0 Ethernet controller: Intel Corporation Ethernet Controller 10-Gigabit X540-AT2 (rev 01)
+   01:00.1 Ethernet controller: Intel Corporation Ethernet Controller 10-Gigabit X540-AT2 (rev 01)
+   ```
+   In our case, both 01:00.0 and 01:00.1 are the two ports on our x540-AT2 card 
+   that we'll use.  You can use `lshw` command to get further details on the 
+   controller and verify it indeed supports SR-IOV.
+
+3. Check how many VFs we can create:
+
+   ```
+   $ cat /sys/bus/pci/devices/0000\:01\:00.0/sriov_totalvfs
+   63
+   $ cat /sys/bus/pci/devices/0000\:01\:00.1/sriov_totalvfs
+   63
+   ```
+
+4. Create VFs:
+
+   Create virtual functions by editing `sriov_numvfs`. In our example, we create
+   one per physical device.  Note, this eliminates the usefulness 
+   of SR-IOV, and is just done for simplicity in this example so you needn't
+   look at 128 virtual devices.
+
+   ```
+   # echo 1 > /sys/bus/pci/devices/0000\:01\:00.0/sriov_numvfs
+   # echo 1 > /sys/bus/pci/devices/0000\:01\:00.1/sriov_numvfs
+   ```
+
+5. Verify that these were added to the host:
+
+   ```
+   # lspci | grep Ethernet | grep Virtual
+   02:10.0 Ethernet controller: Intel Corporation X540 Ethernet Controller Virtual Function (rev 01)
+   02:10.1 Ethernet controller: Intel Corporation X540 Ethernet Controller Virtual Function (rev 01)
+   ```
+
+6. Assign MAC address to each VF
+
+   Depending on the NIC being used, you may need to explicitly set the MAC
+   address for the VF device in order to guarantee that the address is 
+   consistent on the host and when passed through to the guest.
+   For example,
+
+   ```
+   $ ip link set <pf> vf <vfidx> mac <fake MAC address>
+   ```
+
+## Example: Launch a Clear Container using SR-IOV
+
+1. Build and start SR-IOV plugin
+
+   This assumes you already have `GOPATH` set in your environment.
+   ```
+   $ sudo mkdir -p /etc/docker/plugins
+   $ sudo cp $GOPATH/src/github.com/clearcontainers/sriov/sriov.json /etc/docker/plugins/
+   $ cd $GOPATH/src/github.com/clearcontainers/sriov
+   $ go build
+   $ sudo ./sriov &
+   ```
+
+2. Create docker network
+
+   To create the SR-IOV based docker network, a subnet, vlanid, and physical interface are required.
+
+   ```
+   $ sudo docker network create -d sriov --internal --opt pf_iface=enp1s0f0 --opt vlanid=100 --subnet=192.168.0.0/24 vfnet
+
+   E0505 09:35:40.550129    2541 plugin.go:297] Numvfs and Totalvfs are not same on the PF - Initialize numvfs to totalvfs
+   ee2e5a594f9e4d3796eda972f3b46e52342aea04cbae8e5eac9b2dd6ff37b067
+   ```
+
+3. Start containers and test connectivity
```

Comment:
Lines 247-251 rewrite: 

Start the containers and test their connectivity: 

NOTE: insert sudo docker run command immediately following numbered step

The previous command starts a container, which uses SR-IOV.